packages/frontend/core/src/hooks/current/use-current-sync-engine.ts

```diff
@@ -0,0 +1,13 @@
+import type { SyncEngine } from '@affine/workspace/providers';
+
+import { useCurrentWorkspace } from './use-current-workspace';
+
+export function useCurrentSyncEngine(): SyncEngine | undefined {
+  const [workspace] = useCurrentWorkspace();
+  // FIXME: This is a hack to get the sync engine, we need refactor this in the future.
+  const syncEngine = (
+    workspace.blockSuiteWorkspace.providers[0] as { engine?: SyncEngine }
```

Comment:
the provider must exists right?

Reply:
Affine public workspace(shared page) currently does not have a SyncEngine

Reply:
we could make a standalone "sync" peer for it, with pull `download` and push do nothing

Reply:
The logic here is very different. The shared page is implemented by manually download the doc from the `react-router` loader.

There is currently no good way to tell SyncEngine not to load the entire workspace.